llvm/lib/CodeGen/CodeGenTargetMachineImpl.cpp

```diff
@@ -132,8 +133,10 @@ bool CodeGenTargetMachineImpl::addAsmPrinter(PassManagerBase &PM,
                                              MCContext &Context) {
   Expected<std::unique_ptr<MCStreamer>> MCStreamerOrErr =
       createMCStreamer(Out, DwoOut, FileType, Context);
-  if (auto Err = MCStreamerOrErr.takeError())
+  if (auto Err = MCStreamerOrErr.takeError()) {
+    Context.reportError(SMLoc(), toString(std::move(Err)));
```

Comment:
---
(nit) Could avoid moving the Error twice:
```suggestion
  if (!MCStreamerOrErr) {
    Context.reportError(SMLoc(), toString(MCStreamerOrErr.takeError()));
```